lib/rubocop/cop/mixin/non_modifier_if_then.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for checking if an `if` node is in nonmodifier form with implicit `then`.
+    module NonModifierIfThen
+      NON_MODIFIER_THEN = /then\s*(#.*)?$/.freeze
+      NON_MODIFIER_THEN_SAME_LINE = /if.+then/.freeze
+
+      def non_modifier_if_then?(node, same_line: false)
+        return false unless node.if_type?
+
+        (same_line ? NON_MODIFIER_THEN_SAME_LINE : NON_MODIFIER_THEN)
+          .match?(node.loc.begin&.source_line)
```

Comment:
Location information should be used instead of regular expressions.

Reply:
I tried `same_line?(node.children[1], node)` approach to match `then <cond>` case (when `same_line` is true)